### DISCUSSION TITLE
WAS Node Fail - ComfyUI "reds-out" WAS Nodes

### DISCUSSION THREAD

**Discussion by Torcelllo** [WASasquatch/was-node-suite-comfyui#319] [Q&A]
Nothing fixes this, sadly!

![WAS Node fail](https://github.com/WASasquatch/was-node-suite-comfyui/assets/114806454/816d5bdd-c0d2-4f88-8fe9-178f4df4f0fc)

**Comment by Torcelllo** [comment]
I went to the folder - performed git pull - said all was up-to-date!

**Reply by stephantual** [reply]
Did you try to uninstall/reinstall? I get this sometimes when one of my 3901391 nodes decides to act up.

**Reply by WASasquatch** [reply]
Commonly with WAS-NS is installing it first and then bunch of other stuff updating dependencies that needs to be pinned.

**Reply by Torcelllo** [reply]
Uninstall/reinstall did not fix

**Comment by WASasquatch** [comment]
You would have errors in your console. None of this helps diagnose anything, _especially_ from a third party installer.

**Reply by Torcelllo** [reply]
What is my console - u mean ComfyUI?

**Reply by stephantual** [reply]
He means the terminal from which you started comfy ui. there should be an error log there.

**Reply by Torcelllo** [reply]
IMPORT FAILED

**Reply by Torcelllo** [reply]
For some reason, several of my Custom Nodes are simply IMPORT FAILED - including WAS

**Reply by Torcelllo** [reply]
Cannot import B:\ComfyUI\custom_nodes\was-node-suite-comfyui module for custom nodes: Command '['C:\\Users\\john_\\AppData\\Local\\Programs\\Python\\Python310\\python.exe', '-s', '-m', 'pip', 'freeze']' returned non-zero exit status 2

**Reply by stephantual** [reply]
Yeah i upgraded today to the latest comfy and when i fetched the updates from the manager, 4 failed including WAS. I checked and they are functional though - i'm sorry to hear it's not the case for you. I'm not sure how i can help - it's difficult without being in front of the machine at this point.

**Comment by WASasquatch** [comment]
My biggest advice is to not use Manager for WAS-NS. It's cool, but has no concept of order things should be installed. WAS-NS should be installed last because it will downgrade modules to use older code. For example, I support BLIP while transformers has abandoned it for BLIP2 which no one can run right in ComfyUI on consumer GPUs and negligent to move to and focus on.

**Reply by kyrvas48** [reply]
Hello @WASasquatch - I like your handle; Happy New Year! 
I don't have an answer but do have additional information and a Mac Terminal Log. Also, WAS node types failed to load BEFORE using the Manager to “Install Missing Custom Nodes”. 

**WAS “node types not found” Error in ComfyUI workflow**

Hardware: Mac M1 Mini silicon
Software: Mac Terminal

Installed was-node-suite-comfyui into: 

/ComfyUI/custom_nodes/ folder

git clone https://github.com/WASasquatch/was-node-suite-comfyui.git

Mac Terminal log: 

Cloning into 'was-node-suite-comfyui'...
remote: Enumerating objects: 1942, done.
remote: Counting objects: 100% (530/530), done.
remote: Compressing objects: 100% (125/125), done.
remote: Total 1942 (delta 467), reused 440 (delta 405), pack-reused 1412
Receiving objects: 100% (1942/1942), 21.22 MiB | 17.52 MiB/s, done.
Resolving deltas: 100% (1231/1231), done.
 
custom_nodes Folder: 
was-node-suite-comfyui folder = 32 MB

ComfyUI Local Server: 
ComfyUI “node types not found” error after loading a Comfy workflow with WAS nodes: 

When loading the graph, the following node types were not found:
	•	CLIPSeg Masking
	•	Mask Dominant Region
	•	Image Resize
	•	Image Filter Adjustments
	•	Image Crop Location
	•	Mask Crop Region
	•	Image Blend
	•	KSampler Cycle
	•	CLIPTextEncode (BlenderNeko Advanced + NSP)
	•	Image Lucy Sharpen
	•	Image Paste Crop
	•	Text Multiline
	•	Image Save
	•	Constant Number
Nodes that have failed to load will show as red on the graph.

Manager:  
“Install Missing Custom Nodes” error result: 

![WAS Node Suite_Disable:Uninstall](https://github.com/WASasquatch/was-node-suite-comfyui/assets/139254762/5dc36049-990e-4ccf-bd1a-fb606809ffd0)

NOTE: PowerNoiseSuite loaded as a separate folder; these nodes, although in the  workflow,  are not listed in ‘failed to load above”. 

Thank you for considering this additional information.

**Comment by WASasquatch** [comment]
Yes you have to restart ComfyUI before Manager even installs the custom
nodes. WAS-NS should be installed last as it needs to downgrade modules. If
it fails to launch *after* installation you can try the force/retry
install.

Additionally, WAS-NS has its own `install.bat` you can run after it's
cloned.

**Reply by kyrvas48** [reply]
Thank you for your response. I believe I followed your instructions but 
on loading a workflow, I get the “node types not found” for WAS. 
force/retry - how do I do this? 

'Install.bat' executes an .exe file which is not compatible with Mac OS.

**Comment by WASasquatch** [comment]
You'll need to re-install the requirements, regardless. So you need to run
the requirements file against your comfy environment
